marklogic-data-hub-central/ui/src/components/explore/entity-specific-sidebar/entity-specific-sidebar.tsx

```diff
@@ -0,0 +1,364 @@
+      } else {
+        setAllSelectedFacets(searchOptions.selectedFacets);
+      }
+      checkFacetRender([]);
+    }
+  }, [greyedOptions]);
+
+  // useEffect(() => {
+  //   setEntitySpecificSearch(searchOptions.query);
+  //   searchOptions.query && setQueryGreyedOptions(searchOptions.query);
+  // }, [searchOptions]);
+
+  const onDateFacetChange = (datatype, facet, value, isNested) => {
+    let updateFacets = {...allSelectedFacets};
+    if (value.length > 1 && value[0]) {
+      updateFacets = {...updateFacets, [facet]: {dataType: datatype, rangeValues: {lowerBound: moment(value[0]).format("YYYY-MM-DD"), upperBound: moment(value[1]).format("YYYY-MM-DD")}}};
+      setAllGreyedOptions(updateFacets);
+      setAllSelectedFacets(updateFacets);
+    } else if (value.length === 0) {
+      clearRangeFacet(facet);
+      clearGreyRangeFacet(facet);
+    }
+  };
+
+  // const updateEntitySpecifSearch = ({target}) => {
+  //   const {value} = target;
+  //   setQueryGreyedOptions(value);
+  //   setEntitySpecificSearch(value);
+  // };
+
+  const updateSelectedFacets = (constraint: string, vals: string[], datatype: string, isNested: boolean, toDelete = false, toDeleteAll: boolean = false) => {
+    let facets = {...allSelectedFacets};
+    let greyFacets = {...greyedOptions.selectedFacets};
+    let type = "";
+    let valueKey = "";
+    let facetName = constraint;
+
+    switch (datatype) {
+    case "xs:string":
+    case "collection": {
+      type = "xs:string";
+      valueKey = "stringValues";
+      break;
+    }
+    case "xs:integer": {
+      type = "xs:integer";
+      valueKey = "rangeValues";
+      break;
+    }
+    case "xs:decimal": {
+      type = "xs:decimal";
+      valueKey = "rangeValues";
+      break;
+    }
+    default:
+      break;
+    }
+
+    if (vals.length > 0) {
+      facets = {
+        ...facets,
+        [facetName]: {
+          dataType: type,
+          [valueKey]: vals
+        }
+      };
+      greyFacets = {
+        ...greyFacets,
+        [facetName]: {
+          dataType: type,
+          [valueKey]: vals
+        }
+      };
+    } else {
+      delete facets[facetName];
+    }
+    if (toDelete) {
+      if (Object.entries(searchOptions.selectedFacets).length > 0 && searchOptions.selectedFacets.hasOwnProperty(constraint)) {
+        clearFacet(constraint, vals[0]);
+      } else if (Object.entries(greyedOptions.selectedFacets).length > 0 && greyedOptions.selectedFacets.hasOwnProperty(constraint)) {
+        clearGreyFacet(constraint, vals[0]);
+      }
+    } else if (toDeleteAll) {
+      clearConstraint(constraint);
+    } else {
+      setAllSelectedFacets(facets);
+      setAllGreyedOptions(greyFacets);
+    }
+  };
+
+
+  const addFacetValues = (constraint: string, vals: string[], dataType: string, facetCategory: string) => {
+    let newAllSelectedfacets = {...allSelectedFacets};
+    let valueKey = "stringValues";
+    // TODO add support for non string facets
```

Comment:
I'm guessing this was copy and paste from the original place that handles the main facets :D
Do we have a ticket to be able to remove this to do?

Reply:
@aniamy I deleted the TODO because exists support for non-string facets. It was a copy-paste error.